stakbucks/230620/b_1629_곱셈/app.js

```diff
@@ -0,0 +1,24 @@
+const fs = require('fs');
+const filePath = process.platform === 'linux' ? '/dev/stdin' : './input.txt';
+let input = fs.readFileSync(filePath).toString().split(' ');
+
+function solution(input) {
+  let [a, b, c] = input.map(BigInt);
+
+  const power = (a, b) => {
+    // 0 제곱하면 1
+    if (b === BigInt(0)) return BigInt(1);
+
+    // a^b=a^(b/2)*a^(b/2) 이용
+    const half = power(a, BigInt(parseInt(b / BigInt(2))));
+    if (b % BigInt(2)) {
+      // b가 홀수인 경우
+      return (half * half * (a % c)) % c;
+    }
```

Comment:
```suggestion
    if (b === 0n) return BigInt(1);

    // a^b=a^(b/2)*a^(b/2) 이용
    const half = power(a, BigInt(parseInt(b / BigInt(2))));
    if (b % 2n) {
      // b가 홀수인 경우
      return (half * half * (a % c)) % c;
    }
```
요런식으로 BigInt를 좀 더 간결하게 표현할 수 있다고 하네요 ㅎㅎ

Reply:
맞아요 저도 더 찾아보다 알았어요. BigInt 대신 n을 붙이는 게 더 가독성이 좋고 간결한 거 같아요